### ISSUE TITLE
add windows explorer intergration

### ISSUE THREAD

**Issue by prabirshrestha** [neovim/neovim#6649]
Would be good to have something like `nvim --register` so it can register OS integrations (Vim has install.exe). Most likely this would be a neovim-qt feature.

Having `Edit in NeoVim` would be the minimal requirement but better if it can also figure out all open instances of neovim and have an ability to open in that particular instance like how vim does. In this example I have two separate instances of vim already opened - setenv.cmd and .vimrc.

![image](https://cloud.githubusercontent.com/assets/287744/25627539/becae23c-2f17-11e7-8d5f-a501f0889a5e.png)

**Comment by justinmk** [comment]
It can be a command, e.g. `:Foo`. Don't need a command line option.

**Comment by ry0id** [comment]
Was this ever added?

**Comment by justinmk** [comment]
@ry0id No, that's why the issue is still open.

**Comment by AzP** [comment]
This is one of the most pressing issues with neovim-qt for me, since I'm a daily windows user (at work), and have tried to completely migrate to neovim-qt from gvim.
The fact that "open with neovim-qt" doesn't work is a huge blocker. This would be a very big improvement in the day-to-day usage.

**Comment by nadya-p** [comment]
As a workaround, you can add the context menu entry directly in Windows Registry. I've named the entry as "y Open with nvim" in order to have Win+Y shortcut accessible (Windows takes the first letter as a keyboard shortcut). Win+V is already associated with "Restore previous versions".

You'd need to adjust the path to your neovim-qt executable in the code
```Windows Registry Editor Version 5.00
[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\*\shell\y Open with nvim]
[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\*\shell\y Open with nvim\command]
@="C:\\Neovim\\bin\\nvim-qt.exe \"%1\""
[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\Directory\shell\y Open with nvim]
[HKEY_LOCAL_MACHINE\SOFTWARE\Classes\Directory\shell\y Open with nvim\command]
@="C:\\Neovim\\bin\\nvim-qt.exe \"%1\""
```

save it as .reg file and and then apply it to registry.

**Comment by xylix** [comment]
Gvim uses --remote for implementing the "open file in existing vim session" feature (source: https://vim.fandom.com/wiki/Open_file_in_already_running_vim_from_elsewhere ). Does Neovim support a feature like that?

Also, would a windows :Install command that has _only_ the optional registry edits to add the "open with nvim" get accepted as a PR? I'm somewhat interested in implementing that but I don't personally need the "open in existing vim" kind of features.

Would be good to have a list of features that users would want as opt-ins or defaults in the :Install command. This thread has 0 info about possible linux features for :Install. 

@justinmk Do you happen to know of any other related issues that have been made after this was last updated?

**Comment by xylix** [comment]
Some feature ideas

- Bind things to :Install in init.vim (so users wouldn't need to run both :Install and their possible plugin's PlugInstall etc, instead could "append" it all to :Install).
- Uninstall command to undo reg edits etc system wide changes made by :Install

**Comment by justinmk** [comment]
> would a windows :Install command that has _only_ the optional registry edits to add the "open with nvim" get accepted as a PR?

Sure, anything that gets it started. 

> Would be good to have a list of features that users would want as opt-ins or defaults in the :Install command

Not necessary. Opt-out means "don't run `:Install`", at least for this first phase :)



> * Bind things to :Install in init.vim (so users wouldn't need to run both :Install and their possible plugin's PlugInstall etc, instead could "append" it all to :Install).

I don't follow. How is that different than just putting multiple commands in init.vim ? 

> * Uninstall command to undo reg edits etc system wide changes made by :Install

Yes that's probably necessary.

**Comment by w-tr** [comment]
@justinmk and @xylix, I'm somewhat aloof to the ":Install". In my cut of nvim when I do ":h Install" or ":h :Install" I get nothing.

Am I correct in thinking that when first mentioned in 2017 to present this :Install is just a concept?

**Comment by itsmeboy0** [comment]
#17390 was merged some time ago, which means Windows installers already exist, but there isn't any explorer integration yet.

Ping @RenFraser

**Comment by justinmk** [comment]
Yeah, this is low priority now, but `:Install` would still be useful because (1) it's easier to inspect the script to see what it does, (2) it's useful for portable builds.

**Comment by gbaychev** [comment]
Hello everybody,
I just want to ask you: are you sure about this `:Install` thing? 

I would argue that things like creating registry entries for windows integration, shortcuts, etc - which is the basic setup of the application - is logically the job of the installer. And more importantly, the installer is going to remove the registry entries and whatever integration was set up, when neovim is uninstalled. 

Otherwise, if you do `:Install` on a portable installation and delete the installation, you're left with a non-functioning hole in your windows context menu. You're going to need to either remove the registry entries per hand or we must provide a `:Uninstall` command, which the user is going to forget every single time.

So what is your opinion?
